pkg/virt-launcher/virtwrap/network/podinterface_test.go

```diff
@@ -419,11 +433,14 @@ var _ = Describe("Pod Network", func() {
 					},
 				}
 				vmi := newVMI("testnamespace", "testVmName")
-				podnic := podNICImpl{}
-				err := podnic.PlugPhase1(vmi, iface, net, "fakeiface", pid)
+				podnic := newPodNIC(vmi)
+				podnic.iface = iface
+				podnic.network = net
+				podnic.podInterfaceName = "fakeiface"
+				err := podnic.PlugPhase1(pid)
```

Comment:
I don't understand why instead of having https://github.com/kubevirt/kubevirt/pull/5062/commits/af0df80aa305e60b16f9fa902be40c69b506fd19#diff-7ca01cbb567927ef22a4ae7dfb82dcc71ccd5b3d5d3980eba72d5389bfc15d0cR96 you don't have a function that receives the iface, the network, and the podInterfaceName as parameters and just returns a new nic w/ those attributes.

Reply:
90% of the test use the default values, if they one to change some of them just use the default and change it.

Reply:
I would prefer to have *one* way of creating the pod nics in the tests versus mutating the one podNIC that we "know" how to create. 

The motivation being if I have one way of creating them that just works, means that I won't have to reason about whatever is happening on the 10% of times that it doesn't use the default. Or better yet, why have it nice and easy to reason about 90% of the times when you can have it always.

But again, I can't force you to think like I do, nor I want to block this PR based on a minor detail.